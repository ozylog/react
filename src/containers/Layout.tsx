import React from 'react';
import styled from 'styled-components';
import { useContextSelector } from 'use-context-selector';
import { FiMenu } from 'react-icons/fi';

import LayoutContext from './../contexts/LayoutContext';

const Header = styled.header`
  background: #ccc;
  text-align: center;
  padding: 20px;
`;

const Container = styled.div`
  padding: 20px;
`;

const Title = styled.h1`
  font-size: 23px;
  font-weight: 500;
`;

interface Props {
  title: string;
}

function Layout({ title , children }: React.PropsWithChildren<Props>): React.ReactElement {
  const showMenu = useContextSelector(LayoutContext, (state) => state.showMenu);

  return (
    <>
      <Header>
        <FiMenu />
        <Title>
          {title}
        </Title>
      </Header>
      <Container>
        {children}
      </Container>
    </>
  );
}

export default React.memo(Layout);

