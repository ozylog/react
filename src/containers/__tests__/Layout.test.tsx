import renderer from 'react-test-renderer';
import Layout from '../Layout';
import 'jest-styled-components';

describe('<Layout />', () => {
  it('should render Layout component when title = Register card form', () => {
    const component = renderer.create(
      <Layout title='Register card form' />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('should render Layout component when title = Menu', () => {
    const component = renderer.create(
      <Layout title='Menu' />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
});
