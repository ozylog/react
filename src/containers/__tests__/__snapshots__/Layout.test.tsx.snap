// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> should render Layout component when title = Menu 1`] = `
Array [
  .c0 {
  background: #ccc;
  text-align: center;
  padding: 20px;
  position: relative;
}

.c2 {
  font-size: 23px;
  font-weight: 500;
}

.c1 {
  position: absolute;
  top: 25px;
  left: 15xp;
  font-size: 20px;
  cursor: pointer;
}

<header
    className="c0"
  >
    <div
      className="c1"
      onClick={[Function]}
    >
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <line
          x1="3"
          x2="21"
          y1="12"
          y2="12"
        />
        <line
          x1="3"
          x2="21"
          y1="6"
          y2="6"
        />
        <line
          x1="3"
          x2="21"
          y1="18"
          y2="18"
        />
      </svg>
    </div>
    <h1
      className="c2"
    >
      Menu
    </h1>
  </header>,
  .c0 {
  padding: 20px;
}

<div
    className="c0"
  />,
]
`;

exports[`<Layout /> should render Layout component when title = Register card form 1`] = `
Array [
  .c0 {
  background: #ccc;
  text-align: center;
  padding: 20px;
  position: relative;
}

.c2 {
  font-size: 23px;
  font-weight: 500;
}

.c1 {
  position: absolute;
  top: 25px;
  left: 15xp;
  font-size: 20px;
  cursor: pointer;
}

<header
    className="c0"
  >
    <div
      className="c1"
      onClick={[Function]}
    >
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <line
          x1="3"
          x2="21"
          y1="12"
          y2="12"
        />
        <line
          x1="3"
          x2="21"
          y1="6"
          y2="6"
        />
        <line
          x1="3"
          x2="21"
          y1="18"
          y2="18"
        />
      </svg>
    </div>
    <h1
      className="c2"
    >
      Register card form
    </h1>
  </header>,
  .c0 {
  padding: 20px;
}

<div
    className="c0"
  />,
]
`;
